openedx/core/djangoapps/appsembler/api/v1/views.py

```diff
@@ -212,9 +212,15 @@ def create(self, request):
                 request=request,
                 params=data,
             )
-            # set the user as active if password is provided
-            # meaning we don't have to send a password reset email
-            user.is_active = password_provided
+            if not password_provided:
+                # if the password is not provided, keep the user inactive until
+                # the password is set.
+                user.is_active = False
+            else:
+                # if send_activation_email is True, we want to keep the user
+                # inactive until the email is properly validated. If the param
+                # is False, we activate it.
+                user.is_active = not data['send_activation_email']
```

Comment:
Fixes pep8 issue `./openedx/core/djangoapps/appsembler/api/tests/test_registration_api.py:120:43: E712 comparison to False should be 'if cond is False:' or 'if not cond:'`

```suggestion
            if password_provided:
                # if send_activation_email is True, we want to keep the user
                # inactive until the email is properly validated. If the param
                # is False, we activate it.
                user.is_active = not data['send_activation_email']
            else:
                # if the password is not provided, keep the user inactive until
                # the password is set.
                user.is_active = False
```

Reply:
Done! Thanks @OmarIthawi !